app/views/play/modal/PlayItemsModal.coffee

```diff
@@ -212,6 +214,45 @@ module.exports = class PlayItemsModal extends ModalView
       @$el.one 'click', (e) ->
         button.removeClass('confirm').text($.i18n.t('play.unlock')) if e.target isnt button[0]
 
+  onSellButtonClicked: (e) ->
+    e.stopPropagation()
+    button = $(e.target).closest('button')
+    item = @idToItem[button.data('item-id')]
+    
+    # Prevent further action if the item cannot be sold
+    if !item.sellable()
+      return
+    
+    if button.hasClass('confirm')
+      # Sell logic here
+      sale = Sale.makeFor(item)
+      sale.save()
+    
+      #- set local changes to mimic what should happen on the server...
+      sales = me.get('sales') ? {}
+      sales.items ?= []
+      sales.items.push(item.get('original'))
+    
+      purchased = me.get('purchased') ? {}
+      purchased.items ?= []
+      _.pull(purchased.items, item.get('original'))
+      me.set('purchased', purchased)
+    
```

Comment:
Oh yeah, and the current implementation doesn't wait for the network request to end, but this one should, as described in the form recipe. Basically, the button should say 'selling', be disabled, and then re-enabled when the network request has succeeded.

https://github.com/codecombat/codecombat/wiki/Add-a-form#submitting-data

The way forms currently behave is inconsistent across the site, but we should generally move toward these implementations and behaviors.